snafu/trex_wrapper/scripts/run_simple_burst

```diff
@@ -0,0 +1,221 @@
+            ansi_escape = re.compile(r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])')
+            if isinstance(msg, (str, bytes)):
+                msg = ansi_escape.sub('', msg)
+                self.msg += msg
+            if newline:
+                log.debug(self.msg)
+                self.msg = ''
+
+    def _flush (self):
+        pass
+
+clog = CustomLogger("debug")
+
+
+class STLS1(object):
+    def __init__ (self, num_port, size, rate, duration, stream, verbose="debug"):
+        self.client = STLClient(verbose_level=verbose, logger=clog)
+        self.size = size
+        self.pg_id = 0
+        self.num_streams = stream
+        self.force_exit = False
+        self.num_ports = num_port
+        self.ports = [i for i in range(num_port)]
+        self.rate = rate
+        self.duration = duration
+        signal.signal(signal.SIGINT, self.signal_handler)
+
+    def create_vm(self, direction=0):
+        ip_range = {'src': {'start': "10.0.0.1", 'end': "10.0.0.254"},
+                    'dst': {'start': "8.0.0.1",  'end': "8.0.0.254"}}
+
+        if (direction == 0):
+            src = ip_range['src']
+            dst = ip_range['dst']
+        else:
+            src = ip_range['dst']
+            dst = ip_range['src']
+
+        vm = [
+            # src  
+            STLVmFlowVar(name="src",min_value=src['start'],max_value=src['end'],size=4,op="inc"),
+            STLVmWrFlowVar(fv_name="src",pkt_offset= "IP.src"),
+
+            # dst
+            STLVmFlowVar(name="dst",min_value=dst['start'],max_value=dst['end'],size=4,op="inc"),
+            STLVmWrFlowVar(fv_name="dst",pkt_offset= "IP.dst"),
+
+            # checksum
+            STLVmFixIpv4(offset = "IP")
+            ]
+        return vm
+
+
+    def create_stream (self, dest_mac=None, port=0):
+        size = self.size - 4; # HW will add 4 bytes ethernet CRC
+
+        ethr_base = Ether(dst=dest_mac) / IP() / UDP()
+
+        pad = max(0, size - len('ether')) * 'x'
+        vm = self.create_vm(port)
+        pkt = STLPktBuilder(pkt=ethr_base/pad, vm=vm)
+
+        streams = []
+        for pg_id_add in range(0, self.num_streams):
+            streams.append(STLStream(packet = pkt, mode = STLTXCont(pps=1), flow_stats = STLFlowStats(pg_id = self.pg_id + pg_id_add)))
+        return streams
+
+    def add_streams(self, port, dest_mac):
+        if dest_mac:
+            log.info("create stream for port %s with dest mac %s" % (port, dest_mac))
+        new_stream = self.create_stream(dest_mac, port)
+        self.client.add_streams(new_stream, [port])
+
+    def init_streams(self):
+        macs = os.getenv("DEST_MAC_LIST").split(',') or os.getenv("dest_mac_list").split(',')
+        log.info("destination macs available")
+        for idx in range(0, self.num_ports):
+            self.add_streams(idx, macs[idx])
+
+    def collect_stats(self, duration):
+        output = []
+        current_ts = time.time()
+
+        for i in range(0,duration):
+            time.sleep(1)
```

Comment:
lets allow the user to provide a collection interval... and add this to the metadata. Long running zero drop tests could generate a lot of data... if we need that sort of granularity, that is fine, but there might be a use to allow a 10 second interval.

Reply:
Yes, it make sense to make this configurable based on the test load.